### PR TITLE
separate network resource and ec2 instance creation using data block

### PR DESCRIPTION
**Pull request by benlee0423**
This change is related to https://github.com/AlabamaWaterInstitute/CloudInfra/issues/9.

1. Separate TF script into two separate directory, network related resource creation and ec2 instance creation.
2. Clean up and refactor existing TF code. 
3. As a user, run TF code in user directory.

TODO: find a way to mount S3 bucket into the docker container.

### CODE REVIEW

**Review by arpita0911patel — COMMENTED** [review]
I checked out the PR and tested on my local machine, it worked great.

**Comment by benlee0423** [comment]
@ZacharyWills Please review/approve it when you have time.

File: README.md [hunk 1]
```diff
@@ -18,7 +18,9 @@ This is now done by Github Actions automatically, on push to main branch or when
 
 ## Terraform Usage
 
-To run the Terraform configuration, follow these steps:
+** Note that terraform configuration in network directory will setup initial VPC and subnets, only runs once by administrator.
+To run the Terraform configuration as a user, follow these steps in user directory under terraform:
+As a pre-requisite step, create a key-pair in AWS EC2 console, and place the key in user directory.
```
Comment (ZacharyWills) [inline README.md:23]: Might want to list the directory here as the users might not have a lot of experience with AWS.

**Review by ZacharyWills — CHANGES_REQUESTED** [review]
Looks good, add the location for the AWS keys and we can merge.

File: README.md (hunk 1 above)
Comment (benlee0423) [reply README.md:27]: Addressed.

**Review by jameshalgren — APPROVED** [review]

**Pull request merged**